.github/workflows/publish.yml

```
@@ -14,23 +14,43 @@ jobs:
     steps:
 
     - uses: actions/checkout@v2
+        #  with:
+        #  distribution: adopt
+      #  java-version: 11
       with:
         ref: v${{ github.event.inputs.version }}
 
-    # Gradle 7 requires Java 11 to run
+      # Gradle 7 requires Java 11 to run
     - uses: actions/setup-java@v2
       with:
         distribution: 'zulu'
         java-version: '11'
 
-    - name: Publish to Repositories
-      env:
-        ORG_GRADLE_PROJECT_MAPBOX_DOWNLOADS_TOKEN: ${{ secrets.MAPBOX_DOWNLOADS_TOKEN }}
-        ORG_GRADLE_PROJECT_MAPBOX_ACCESS_TOKEN: ${{ secrets.MAPBOX_ACCESS_TOKEN }}
-        ORG_GRADLE_PROJECT_ABLY_API_KEY: ${{ secrets.ABLY_API_KEY }}
-        ORG_GRADLE_PROJECT_GOOGLE_MAPS_API_KEY: ${{ secrets.GOOGLE_MAPS_API_KEY }}
-        ORG_GRADLE_PROJECT_GITHUB_ACTOR: ${{ github.actor }}
-        ORG_GRADLE_PROJECT_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      run: |
-        echo "Publishing version ${{ github.event.inputs.version }}..."
-        ./gradlew publish
+  - name: Release build
+      # assembleRelease for all modules, excluding non-library modules
+      run: ./gradlew assembleRelease -x :android-test-common:assembleRelease -x :test-common:assembleRelease -x
+      :integration-testing-app:assembleRelease -x :publishing-example-app:assembleRelease -x
+      :publishing-java-testing:assembleRelease -x :subscribing-example-app:assembleRelease -x :subscribing-java-testing:assembleRelease
+      - name: Source jar and dokka
+        run: ./gradlew androidSourcesJar javadocJar
+      - name: Publish to MavenCentral
+        run: ./gradlew publishReleasePublicationToSonatypeRepository
+        env:
+          ORG_GRADLE_PROJECT_MAPBOX_DOWNLOADS_TOKEN: ${{ secrets.MAPBOX_DOWNLOADS_TOKEN }}
+          ORG_GRADLE_PROJECT_MAPBOX_ACCESS_TOKEN: ${{ secrets.MAPBOX_ACCESS_TOKEN }}
+          ORG_GRADLE_PROJECT_ABLY_API_KEY: ${{ secrets.ABLY_API_KEY }}
+          ORG_GRADLE_PROJECT_GOOGLE_MAPS_API_KEY: ${{ secrets.GOOGLE_MAPS_API_KEY }}
+          # Following two may not be needed anymore
+          ORG_GRADLE_PROJECT_GITHUB_ACTOR: ${{ github.actor }}
+          ORG_GRADLE_PROJECT_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          OSSRH_USERNAME: ${{ secrets.OSSRH_USERNAME }}
```

Aligned with the environment above I would suggest also using the lesser-known `ORG_GRADLE_PROJECT_` prefix for all of these also, to set Gradle properties automatically. e.g. for this line:

```suggestion
          ORG_GRADLE_PROJECT_OSSRH_USERNAME: ${{ secrets.OSSRH_USERNAME }}
```

This will result in much less complicated Gradle scripts downstream.

---

https://github.com/ably/ably-asset-tracking-android/pull/499/commits/24e2af5671bbeb884ac832ebdeffd3df9e5872ab

---

I would not have expected adopting this scheme for properties to be as simple as _just_ https://github.com/ably/ably-asset-tracking-android/commit/24e2af5671bbeb884ac832ebdeffd3df9e5872ab. There must be other changes required.

---

https://github.com/ably/ably-asset-tracking-android/pull/499/commits/0e58138d69daaf68af20c37b31e8d7940c549bd6